shared/src/tx_pool/commit_txs_scanner.rs

```diff
@@ -0,0 +1,210 @@
+use crate::tx_pool::{
+    proposed::ProposedPool,
+    types::{TxEntry, TxModifiedEntries},
+};
+use ckb_core::transaction::ProposalShortId;
+use ckb_core::Cycle;
+use std::cmp::max;
+use std::collections::HashSet;
+
+/// node will give up to package more txs after MAX_CONSECUTIVE_FAILED
+const MAX_CONSECUTIVE_FAILED: usize = 500;
+
+/// find txs to package into commitment
+pub struct CommitTxsScanner<'a> {
+    proposed_pool: &'a ProposedPool,
+    entries: Vec<TxEntry>,
+    // modified entries, after put a tx into block,
+    // the scores of descendants txs should be updated,
+    // these modified entries is stored in modified_entries.
+    // in each loop,
+    // we pick tx from modified_entries and pool to find the best tx to package
+    modified_entries: TxModifiedEntries,
+    // txs that packaged in block
+    fetched_txs: HashSet<ProposalShortId>,
+}
+
+impl<'a> CommitTxsScanner<'a> {
+    pub fn new(proposed_pool: &'a ProposedPool) -> CommitTxsScanner<'a> {
+        CommitTxsScanner {
+            proposed_pool,
+            entries: Vec::new(),
+            modified_entries: TxModifiedEntries::default(),
+            fetched_txs: HashSet::default(),
+        }
+    }
+    /// find txs to commit, return TxEntry vector, total_size and total_cycles.
+    pub fn transactions_to_commit(
```

Comment:
This function is still too complex, but I'm OK to approve this PR, since now it is in a single module and is easy to refactor later.